### PR TITLE
Create a tool manifest

### PR DESCRIPTION
**Pull request by dougbu**
- see dotnet/dnceng#573
- trim-assets-version.ps1 is the only place we use `dotnet tool install --local`
  - w/o a tool manifest `--local` consistently fails
- this is a reaction to dotnet/arcade@a4c7aafc998e / dotnet/arcade#13979

### CODE REVIEW

**Comment by dougbu** [comment]
I wonder if we _also_ need to fix up trim-assets-version.ps1 in Arcade to run `dotnet new tool-manifest` if a repo using SDL doesn't yet have one. Thoughts❔

/cc @andriipatsula 

----

side note @wtgodbe and @mkArtakMSFT: the 7.0 web templates seem busted&hellip;
``` text
> dotnet new tool-manifest
Warning:
The following templates use the same identity 'Microsoft.Web.Mvc.CSharp.7.0':
   'ASP.NET Core Web App (Model-View-Controller)' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core Web App (Model-View-Controller)' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core Web App (Model-View-Controller)' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Mvc.FSharp.7.0':
   'ASP.NET Core Web App (Model-View-Controller)' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core Web App (Model-View-Controller)' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core Web App (Model-View-Controller)' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.WebApi.CSharp.7.0':
   'ASP.NET Core Web API' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core Web API' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core Web API' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.WebApi.FSharp.7.0':
   'ASP.NET Core Web API' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core Web API' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core Web API' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Worker.Empty.CSharp.7.0':
   'Worker Service' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'Worker Service' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'Worker Service' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Worker.Empty.FSharp.7.0':
   'Worker Service' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'Worker Service' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'Worker Service' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Blazor.Server.CSharp.7.0':
   'Blazor Server App' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'Blazor Server App' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'Blazor Server App' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Blazor.Wasm.CSharp.7.0':
   'Blazor WebAssembly App' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'Blazor WebAssembly App' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'Blazor WebAssembly App' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Blazor.Server.Empty.CSharp.7.0':
   'Blazor Server App Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'Blazor Server App Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'Blazor Server App Empty' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Blazor.Wasm.Empty.CSharp.7.0':
   'Blazor WebAssembly App Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'Blazor WebAssembly App Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'Blazor WebAssembly App Empty' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Empty.CSharp.7.0':
   'ASP.NET Core Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core Empty' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Empty.FSharp.7.0':
   'ASP.NET Core Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core Empty' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core Empty' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Grpc.Service.CSharp.7.0':
   'ASP.NET Core gRPC Service' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core gRPC Service' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core gRPC Service' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.Razor.Library.CSharp.7.0':
   'Razor Class Library' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'Razor Class Library' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'Razor Class Library' will be used. To resolve this conflict, uninstall the conflicting template packages.
Warning:
The following templates use the same identity 'Microsoft.Web.RazorPages.CSharp.7.0':
   'ASP.NET Core Web App' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
   'ASP.NET Core Web App' from 'Microsoft.DotNet.Web.ProjectTemplates.7.0::7.0.0-rc.2.22476.2'
The template from 'ASP.NET Core Web App' will be used. To resolve this conflict, uninstall the conflicting template packages.
The template "Dotnet local tool manifest file" was created successfully.
```

**Comment by riarenas** [comment]
I would like to hear from @andriipatsula if this is expected. Will other repositories need to react in this same way? If they do, we should send proactive communication, etc etc.

**Comment by dougbu** [comment]
> I would like to hear from @andriipatsula if this is expected. Will other repositories need to react in this same way? If they do, we should send proactive communication, etc etc.

since he's likely offline, I suggest we unbreak our official builds and get this in. despite the lack of required approvers here, I want to know at least one person agrees w/ me

**Comment by riarenas** [comment]
I disagree. This would mean tomorrow repos would get a potentially broken Arcade. The whole point of Arcade-validation is to avoid that. If this change is holding up important arcade changes, let's revert https://github.com/dotnet/arcade/pull/13979 instead to unblock the build.

**Review by riarenas — CHANGES_REQUESTED** [review]
I think this needs to be fixed in Arcade and not as a reaction in dependent repositories, or if we do need to react to that Arcade change, we should at least have a lengthy lead up so that repos react to the breaks.

**Comment by dougbu** [comment]
> I disagree. This would mean tomorrow repos would get a potentially broken Arcade. The whole point of Arcade-validation is to avoid that. If this change is holding up important arcade changes, let's revert [dotnet/arcade#13979](https://github.com/dotnet/arcade/pull/13979) instead to unblock the build.

now that I pause long enough to think, you're absolutely right. we _want_ arcade-validation-official broken at the moment. patching things here isn't right.

**Comment by riarenas** [comment]
> we want arcade-validation-official broken at the moment

I think that's a great distinction @dougbu.  We don't care as much to keep this repo's pipelines green as much as we care that we don't let potential breaking changes through. That is different than the model we try to keep for other repos. I appreciate that you figured out the changes that would need to happen as that will be good information for @andriipatsula while he examines the best path forward. For example, do we first want to check the existence of a tool manifest? or do we want to just swallow resulting errors?

**Comment by dougbu** [comment]
specifically on "revert dotnet/arcade#13979" and my earlier suggestion (to auto-create a missing manifest), we could also use `--tool-path` (like source-index-stage1.yml) or `--global` (the darc-init.ps1 default) instead of `--local`

**Pull request closed without merging**
